lisp/ess-r-flymake.el

```diff
@@ -0,0 +1,149 @@
+;;; ess-r-flymake.el --- A ess-r Flymake backend  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2018 J. Alexander Branham (alex DOT branham AT gmail DOT com)
+
+;; This file is not part of GNU Emacs.
+
+;; This is free software; you can redistribute it and/or modify it under
+;; the terms of the GNU General Public License as published by the Free
+;; Software Foundation; either version 3, or (at your option) any later
+;; version.
+;;
+;; This is distributed in the hope that it will be useful, but WITHOUT
+;; ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+;; FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+;; for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs; see the file COPYING.  If not, write to the
+;; Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+;; MA 02110-1301 USA.
+
+
+;;; Commentary:
+;; Flymake is the built-in Emacs package that supports on-the-fly
+;; syntax checking.  This file adds support for this in R-mode by
+;; relying on the lintr package, available on CRAN and currently
+;; hosted at https://github.com/jimhester/lintr.
+
+;; It is enabled by default.
+
+;;; Code:
+
+(eval-when-compile
+  (require 'cl-lib))
+
+(defcustom ess-r-flymake-linters "default_linters"
+  "Default linters to use.
+
+See \"lintr::with_defaults\" for how to customize this."
+  :group 'ess-R
+  :type 'string)
+
+(defcustom ess-r-flymake-lintr-cache t
+  "If non-nil, cache lintr results."
+  :group 'ess-R
+  :type 'string)
+
+(defvar-local ess-r--flymake-proc nil)
+
+(defun ess-r-flymake (report-fn &rest _args)
+  "A Flymake backend for ESS-R modes. Relies on the lintr package.
+
+REPORT-FN is flymake's callback function."
+  (unless (executable-find
+           ;; TODO - check whether lintr package is available
+           "R") (error "Cannot find a suitable R"))
```

Comment:
I am surprised emacs doens't give you warning here. It's a non standard formatting. Would you mind moving the `(error ..` on a separate line?

Reply:
Done